### ISSUE TITLE
Warn for is_numeric()

### ISSUE THREAD

**Issue by GaryJones** [Automattic/VIP-Coding-Standards#384]
## What problem would the enhancement address for VIP?

Most of the time, `is_numeric()` is used to check the value of an argument, such as image dimensions. However, `is_numeric()` [allows](https://3v4l.org/p7CtU) values like `0.01` and `10e3` (and possibly values from other bases), which wouldn't make suitable values.

## Describe the solution you'd like

Warn when `is_numeric()` is used, with a prompt to see if `is_int()` or `is_float()` is a more suitable function.

## What code should be reported as a violation?

```php
is_numeric( $foo );
```

## What code should *not* be reported as a violation?

```php
my_is_numeric( $foo );
CustomNS\is_numeric( $foo );
$bar = 'is_numeric';
is_int( $foo );
is_float( $foo );
```

**Comment by rebeccahum** [comment]
What benefit would this be to VIP specifically? Would a sniff like this better live in the [PHPCS Generic Standards](https://github.com/squizlabs/PHP_CodeSniffer/tree/master/src/Standards/Generic)?

**Comment by GaryJones** [comment]
Yep, good call.
